crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -316,6 +316,17 @@ pub unsafe trait QueryData: WorldQuery {
 /// This must only be implemented for read-only [`QueryData`]'s.
 pub unsafe trait ReadOnlyQueryData: QueryData<ReadOnly = Self> {}
 
+/// A [`QueryData`] whose items can be compared as the original entity.
+///
+/// # Safety
+///
+/// Calling [`EntityBorrow::entity()`](crate::entity::EntityBorrow::entity) on the `Item` must return the `entity` that was passed to [`QueryData::fetch()`].
```

Comment:
```suggestion
/// A [`QueryData`] type that produces a `TrustedEntityBorrow` item 
/// equaling the `Entity` it is addressed by.
///
/// # Safety
/// 
/// [`QueryData::fetch`] must always return an item whose `Entity` 
/// equals the one the function was called with.
/// I.e.: `Self::fetch(fetch, entity, table_row).entity() == entity` always holds.
```

Some clearer language!

Reply:
That does look better!  I want to check that the link syntax there is valid before I click Commit Suggestion, though.